// @rainbow-me/rainbowkit/dist/wallets/walletConnectors/chunk-UEZ25KA5.js
// src/wallets/walletConnectors/tahoWallet/tahoWallet.ts
import { InjectedConnector } from "wagmi/connectors/injected";
var tahoWallet = ({
  chains,
  ...options
}) => ({
  id: "taho",
  name: "Taho",
  iconBackground: "#d08d57",
  iconUrl: async () => (await import("./tahoWallet-2OALNQFU.js")).default,
  downloadUrls: {
    browserExtension: "https://chrome.google.com/webstore/detail/taho/eajafomhmkipbjmfmhebemolkcicgfmd"
  },
  installed: typeof window !== "undefined" && typeof window.tally !== "undefined" && window["tally"] ? true : void 0,
  createConnector: () => {
    return {
      connector: new InjectedConnector({
        chains,
        options: {
          getProvider: () => {
            const getTaho = (tally) => (tally == null ? void 0 : tally.isTally) ? tally : void 0;
            if (typeof window === "undefined")
              return;
            return getTaho(window.tally);
          },
          ...options
        }
      })
    };
  }
});

export {
  tahoWallet
};
