// @lighthouse-web3/sdk/dist/Lighthouse/upload/files/node.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.walk = void 0;
const axios_1 = __importDefault(require("axios"));
const form_data_1 = __importDefault(require("form-data"));
const basePathConvert_1 = __importDefault(require("../../utils/basePathConvert"));
const lighthouse_config_1 = require("../../../lighthouse.config");
async function walk(dir) {
    const { readdir, stat } = eval(`require`)('fs-extra');
    let results = [];
    const files = await readdir(dir);
    for (const file of files) {
        const filePath = `${dir}/${file}`;
        const _stat = await stat(filePath);
        if (_stat.isDirectory()) {
            results = results.concat(await walk(filePath));
        }
        else {
            results.push(filePath);
        }
    }
    return results;
}
exports.walk = walk;
/*
  This function is used to deploy a file to the Lighthouse server.
  It takes the following parameters:
  @param {string} sourcePath - The path of file/folder.
  @param {string} apiKey - The api key of the user.
*/
exports.default = async (sourcePath, apiKey) => {
    const { createReadStream, lstatSync } = eval(`require`)('fs-extra');
    const mime = eval(`require`)('mime-types');
    const token = 'Bearer ' + apiKey;
    const stats = lstatSync(sourcePath);
    try {
        const endpoint = lighthouse_config_1.lighthouseConfig.lighthouseNode + '/api/v0/add';
        if (stats.isFile()) {
            //we need to create a single read stream instead of reading the directory recursively
            const data = new form_data_1.default();
            const mimeType = mime.lookup(sourcePath);
            data.append('file', createReadStream(sourcePath));
            const response = await axios_1.default.post(endpoint, data, {
                withCredentials: true,
                maxContentLength: Infinity,
                maxBodyLength: Infinity,
                headers: {
                    'Content-type': `multipart/form-data; boundary= ${data.getBoundary()}`,
                    Encryption: 'false',
                    'Mime-Type': `${mimeType}`,
                    Authorization: token,
                },
            });
            return { data: response.data };
        }
        else {
            const files = await walk(sourcePath);
            const data = new form_data_1.default();
            const mimeType = mime.lookup(sourcePath);
            files.forEach((file) => {
                //for each file stream, we need to include the correct relative file path
                data.append('file', createReadStream(file), {
                    filepath: (0, basePathConvert_1.default)(sourcePath, file),
                });
            });
            const response = await axios_1.default.post(endpoint, data, {
                withCredentials: true,
                maxContentLength: Infinity,
                maxBodyLength: Infinity,
                headers: {
                    'Content-type': `multipart/form-data; boundary= ${data.getBoundary()}`,
                    Encryption: 'false',
                    'Mime-Type': `${mimeType}`,
                    Authorization: token,
                },
            });
            const temp = response.data.split('\n');
            response.data = JSON.parse(temp[temp.length - 2]);
            /*
              {
                data: {
                  Name: 'flow1.png',
                  Hash: 'QmUHDKv3NNL1mrg4NTW4WwJqetzwZbGNitdjr2G6Z5Xe6s',
                  Size: '31735'
                }
              }
            */
            return { data: response.data };
        }
    }
    catch (error) {
        throw new Error(error.message);
    }
};
